test/studies/diffusion/heat_2d.chpl

```diff
@@ -0,0 +1,45 @@
+// setup simulation parameters
+config const xLen : real = 2,
+             yLen : real = 2,
+             nx = 31,
+             ny = 31,
+             nt = 50,
+             sigma = 0.25,
+             nu = 0.05;
+
+const dx : real = xLen / (nx - 1),
+      dy : real = yLen / (ny - 1),
+      dt : real = sigma * dx * dy / nu;
+
+// define a 2D domain and subdomain
+const dom = {0..<nx, 0..<ny},
+      domInner : subdomain(dom) = dom.expand(-1);
+
+// define an array to hold the solution
+var u : [dom] real;
+
+// setup initial conditions
+u = 1;
+u[
+  (0.5 / dx):int..<(1.0 / dx + 1):int,
+  (0.5 / dy):int..<(1.0 / dy + 1):int
+] = 2;
+
+// run finite difference computation
+var un = u;
+for n in 0..#nt {
```

Comment:
I'd suggest either using `0..<nt` for consistency (to avoid "Why did you use `0..<n` in some places and `0..#n` in others?" questions).  Or, really, I'd probably use `1..nt` since the value of `n` seems to be unused (and I think counting from 1 is more natural when counting iterations like this).

Since the index is unused, you could leave the `n in` off altogether of course, but that may be more confusing than helpful.  If preserving the index, I'd probably rename to `i` since `n` is typically used as a problem size or quantity (unless it's different for this community / wouldn't cause the same pause for this audience).  E.g., in this code, there's no relationship between `n`, `nx`, and `ny` even though it sounds as though there ought to be...